go/channelling/room_manager.go

```diff
@@ -209,9 +209,27 @@ func (rooms *roomManager) Get(roomID string) (room RoomWorker, ok bool) {
 	return
 }
 
+func (rooms *roomManager) isPublicRoom(roomName string) bool {
+	if rooms.PublicRoomNames != nil &&
+		rooms.PublicRoomNames.MatchString(roomName) {
+		return true
+	}
+	return false
+}
+
 func (rooms *roomManager) GetOrCreate(roomID, roomName, roomType string, credentials *DataRoomCredentials, sessionAuthenticated bool) (RoomWorker, error) {
+	isPublic := -1 // Will be evaluated lazily
```

Comment:
I understand your comments, and while the behavior is the same with your changes, you are evaluating the regular expression twice if an anonymous user tries to join a non-existant room. This is something I would like to avoid.